String conversion of Database with process isolation and failing test

Given `processIsolation="true"` in the PHPUnit config and a failing assertion the following error occurs:

```
PHPUnit 5.7.23 by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)

Time: 686 ms, Memory: 6.00MB

There was 1 error:

1) Vendor\Package\Functional\MyTest::doesWhatItShould
PHPUnit_Framework_Exception: PHP Fatal error:  Uncaught Object of class Nimut\TestingFramework\Database\Database could not be converted to string

  thrown in - on line 374

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```

With a bit more debugging it seems like PHPUnit tries to serialize the whole testcase in order to return it to the parent process for displaying the test error.

--- a/src/TestingFramework/TestCase/FunctionalTestCase.php
+++ b/src/TestingFramework/TestCase/FunctionalTestCase.php
@@ -159,14 +159,14 @@ abstract class FunctionalTestCase extends AbstractTestCase
     private $testSystem = null;
 
     /**
-     * Avoid serialization of some properties
+     * Avoid serialization of some properties containing objects
      *
      * @return array
      */
     public function __sleep()
     {
         $objectVars = get_object_vars($this);
-        unset($objectVars['testSystem'], $objectVars['database']);
+        unset($objectVars['database'], $objectVars['testSystem']);
 
         return $objectVars;
     }

--- a/src/TestingFramework/TestCase/FunctionalTestCase.php
+++ b/src/TestingFramework/TestCase/FunctionalTestCase.php
@@ -159,14 +159,14 @@ abstract class FunctionalTestCase extends AbstractTestCase
     private $testSystem = null;
 
     /**
-     * Avoid serlialization of the test system object
+     * Avoid serialization of some properties
      *
      * @return array
      */
     public function __sleep()
     {
         $objectVars = get_object_vars($this);
-        unset($objectVars['testSystem']);
+        unset($objectVars['testSystem'], $objectVars['database']);
 
         return $objectVars;
     }

